src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -803,7 +804,16 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
         debug!("codegen_block({:?}={:?})", bb, data);
 
-        if bx.cx().has_debug() {
+        let tracer = match env::var("YK_TRACER") {
```

Comment:
That's the third time (I think) that this block of code appears somewhere. Should we find some way of putting it in a single place such that it can be called? I appreciate that this might not be easy.

Reply:
There's a comment to this effect somewhere.

The *correct* thing to do is to have a compiler flag, e.g. a `-C tracer=X`. This has other advantages too: if we make the flag tracked, cargo will rebuild things if the flag changes.

Currently we just did the dumbest thing possible to get us going.

Reply:
Here's the comment I mentioned:
https://github.com/softdevteam/ykrustc/pull/41/files#diff-24c5c945888bb0d041e769bfb852de6cR1105

Reply:
I could maybe add a `FIXME` here...?

Reply:
I don't think it's necessary to duplicate the existing `FIXME`.

I'm going to need to know whether we are going to use a flag or an environment for my CI tweaks though.

Reply:
Just to check: we're saying that because it's an environment variable we can't place this in a single function somewhere?

Reply:
> Just to check: we're saying that because it's an environment variable we can't place this in a single function somewhere?

We can. I've started a branch to use a proper flag. It makes the code much nicer. We can cherry pick my commit into this PR once I'm done.

Reply:
Do you want to do that as a separate PR?

Reply:
We can do if you like.

Reply:
Let's do it as a separate PR then.

Reply:
OK.